src/lib/components/identity/identity-details.svelte

```diff
@@ -31,16 +31,27 @@
         revoking = false
     }
 
-    async function handleRoleChange(role: UserRoles): Promise<void> {
-        identity.role = role
-        loading = true
-        caret = false
-        await updateIdentity(identity)
-        loading = false
-        caret=true
+    async function handleRoleChange(userRole: UserRoles): Promise<void> {
+        if (identity.role !== userRole) {
+            loading = true
+            const updateDid = identity
+            updateDid.role = userRole
+            await updateIdentity(updateDid)
+            identity = updateDid
+            searchIdentitiesResults.update(() => {
+                return $searchIdentitiesResults.map((did) => {
+                    if (did.id === identity.id) did = identity
```

Comment:
```suggestion
                    return did.id === identity.id ? identity : did
```

Reply:
I think this map statement is easier to understand and should accomplish the same result